### ISSUE TITLE
WebSocket: Support multi-framing sends

### ISSUE THREAD

**Issue by jhihn** [koush/AndroidAsync#683]
Per RFC 6455 `10.4.  Implementation-Specific Limits` client can choose a maximum frame size. It seems that in 
`private byte[] frame(int opcode, byte [] data, int errorCode, int dataOffset, int dataLength)`
 that there is no maximum frame size applied. The data is shoved into a single frame and delivered this cause problems with clients who do implement a maximum frame size. 

I suggest there be a setMaxFrameSize() function added and the data to be fragmented into multiple frames when it exceeds the maximum frame size.

**Comment by koush** [comment]
Ok thanks, I'll take a look.

**Comment by koush** [comment]
300k is a strange delimiter. dhttps://github.com/koush/AndroidAsync/blob/master/AndroidAsync/src/com/koushikdutta/async/http/HybiParser.java#L361

**Comment by jhihn** [comment]
I created a push request that fixes the issue for me but it is suboptimal.
Check the push requests for the project.

**Comment by jhihn** [comment]
> 300k is a strange delimiter. dhttps://github.com/koush/AndroidAsync/blob/master/AndroidAsync/src/com/koushikdutta/async/http/HybiParser.java#L361

Yes I looked all over it and I did come across that but I wasn't sure what was going on there I compared it to other websocket code and it almost agreed but not enough for me to say that I understood it

**Comment by koush** [comment]
is this a http or https websocket? are you using this library as a server or client websocket?

**Comment by jhihn** [comment]
Unencrypted websocket server. 
Client connects, sends text, server replies with a large text reply.
